lib/pure/strutils.nim

```diff
@@ -84,31 +84,42 @@ from std/private/strimpl import cmpIgnoreStyleImpl, cmpIgnoreCaseImpl, startsWit
 
 
 const
-  Whitespace* = {' ', '\t', '\v', '\r', '\l', '\f'}
+  Whitespace* = {' ', '\t', '\v', '\r', '\n', '\f'}
     ## All the characters that count as whitespace (space, tab, vertical tab,
     ## carriage return, new line, form feed).
 
   Letters* = {'A'..'Z', 'a'..'z'}
     ## The set of letters.
 
+  Vowels* = {'a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U'}
+    ## The set of vowels.
+  
+  Consonants* = Letters - Vowels
+    ## The set of consonants.
+
   Digits* = {'0'..'9'}
     ## The set of digits.
 
-  HexDigits* = {'0'..'9', 'A'..'F', 'a'..'f'}
+  HexDigits* = Digits + {'A'..'F', 'a'..'f'}
     ## The set of hexadecimal digits.
 
-  IdentChars* = {'a'..'z', 'A'..'Z', '0'..'9', '_'}
+  IdentChars* = Letters + Digits + {'_'}
     ## The set of characters an identifier can consist of.
 
-  IdentStartChars* = {'a'..'z', 'A'..'Z', '_'}
+  IdentStartChars* = Letters + {'_'}
     ## The set of characters an identifier can start with.
 
-  Newlines* = {'\13', '\10'}
+  Newlines* = {'\r', '\n'}
     ## The set of characters a newline terminator can start with (carriage
     ## return, line feed).
 
+  Punctuation* = {'!', ',', '.', ':', ';', '?'}
```

Comment:
is there an official source for that? can you add it as a comment in code?

Reply:
This is just a set I made up, without a standard... which is a terrible idea now that you mention it...
Haskell does have an `isPunctuation` function, however, that works properly on all unicode characters.
I will change it to have all ASCII characters that are categorized as punctuation in Haskell.

Reply:
python3 -c 'import string; print(string.punctuation)'
!"#$%&'()*+,-./:;<=>?@[\]^_`{|}~

please also do some research for other popular languages; if there's consensus among those, then adding it is a no-brainer, if not, then we need to think a bit

Reply:
Yeah, Haskell lists characters a little differently...
I changed it to that temporarily, will look up some other languages

Reply:
looks like C and python are off by 1 char:
```nim
proc ispunct(a: cint): cint {.importc, header: "<ctype.h>".}
proc main()=
  var s: set[char]
  for c in char.low..char.high:
    if c.cint.ispunct > 0:
      s.incl c
  echo s
  var s2: set[char]
  for c in """!"#$%&'()*+,-./:;<=>?@[]^_`{|}~""":
    s2.incl c
  echo s2
  echo s == s2
  echo s - s2
  echo s2 - s
main()
```

```
{'!', '\"', '#', '$', '%', '&', '\'', '(', ')', '*', '+', ',', '-', '.', '/', ':', ';', '<', '=', '>', '?', '@', '[', '\\', ']', '^', '_', '`', '{', '|', '}', '~'}
{'!', '\"', '#', '$', '%', '&', '\'', '(', ')', '*', '+', ',', '-', '.', '/', ':', ';', '<', '=', '>', '?', '@', '[', ']', '^', '_', '`', '{', '|', '}', '~'}
false
{'\\'}
{}
```

Reply:
https://github.com/nim-lang/Nim/pull/16708

Reply:
The comment that closed the other PR (https://github.com/nim-lang/Nim/pull/16708#issuecomment-760108509) would be justified if there was no definition or agreement on punctuation, but the fact that both C and python agree (assuming default c locale) is a strong argument.

There's actually a definition of punctuation:
> The standard "C" locale considers punctuation characters all graphic characters (as in isgraph) that are not alphanumeric (as in isalnum).

Applications can always choose to use a different set for their needs, but having an accepted standard makes sense.

Reply:
> The standard "C" locale considers punctuation characters all graphic characters (as in isgraph) that are not alphanumeric (as in isalnum).

Then there is no need to add a Punctuation set that someone will have to look up.how it is specified because `Printable - AlphaNumeric` is much clearer and "self-documenting".